examples/purchase-tester/src/main/java/com/revenuecat/purchasetester/OfferingFragment.kt

```diff
@@ -61,18 +62,33 @@ class OfferingFragment : Fragment(), PackageCardAdapter.PackageCardAdapterListen
     override fun onPurchasePackageClicked(cardView: View, currentPackage: Package, purchaseOption: PurchaseOption?) {
         binding.purchaseProgress.visibility = View.VISIBLE
 
-        Purchases.sharedInstance.purchasePackageOptionWith(
-            requireActivity(),
-            currentPackage,
-            purchaseOption!!, // TODOBC5: Fix API for OTP
-            { error, userCancelled ->
-                if (!userCancelled) {
-                    showUserError(requireActivity(), error)
+        if (purchaseOption == null) {
```

Comment:
trying to think if this would be a typical flow for a customer, but if so, maybe we should have a method with a nullable `purchaseOption`? idk, it does make the intent a little less obvious...

Reply:
Well.... In my mind, ideally we keep the existing purchase methods around in a good enough state that devs can continue using them unless they want to specify the offer themselves, in which case, they will need to use this new `purchaseSubscriptionOption` method. I do think, in order to do that, we should make it so free trials are automatically applied, which might be tricky... But probably not impossible as long as we put some constraints in the documentation.